pkg/source/registry.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package source
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
+
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/logger"
+)
+
+type RegistryOpts struct {
+	CACertPaths      []string
+	RegistryUsername string
+	RegistryPassword string
+	RegistryToken    string
+}
+
+// GetRegistryWithProgress creates new registry instance that provides
+// progress updates to the logger
+func GetRegistryWithProgress(ctx context.Context, registryOpts *RegistryOpts) (*registry.WithProgress, error) {
```

Comment:
As you are not getting an existing resource change the name to `NewRegistryWithProgress` or `CreateRegistryWithProgress`